frontend/simple/views/Join.vue

```diff
@@ -97,36 +97,42 @@
     </section>
 </template>
 <script>
+import {namespace} from '../js/backend/hapi'
 import * as Events from '../../../shared/events'
 import backend from '../js/backend/'
-import * as db from '../js/database'
-import {latestContractState} from '../js/state'
+import { latestContractState } from '../js/state'
 import L from '../js/translations'
+
 export default {
   name: 'Join',
-  mounted () {
-    this.fetchData()
+  async mounted () {
+    let state = await latestContractState(this.$route.query.groupId)
+    console.log(state)
+    if (!state.invitees.find(invitee => invitee === this.$store.state.loggedIn.name)) {
+      // TODO: proper user-facing error
+      console.log(new Error('Invalid Invitation'))
+      this.$router.push({path: '/mailbox'})
+    }
+    for (var i = 0; i < state.members.length; i++) {
+      const contractId = await namespace.lookup(state.members[i])
+      state.members[i] = await latestContractState(contractId)
+    }
+    this.contract = state
   },
   methods: {
-    fetchData: async function () {
-      let state = await latestContractState(this.$route.query.groupId)
-      if (!state.invitees.find(invitee => invitee === this.$store.state.loggedIn)) {
-        console.log(new Error('Invalid Invitation'))
-        await this.$store.dispatch('deleteMail', this.$route.query.inviteHash)
-        this.$router.push({path: '/mailbox'})
-      }
-      this.contract = state
-    },
     accept: async function () {
       try {
         this.errorMsg = null
+        let latest = await backend.latestHash(this.$route.query.groupId)
+        let acceptance = new Events.HashableGroupAcceptInvitation({ username: this.$store.state.loggedIn.name, inviteHash: this.$route.query.inviteHash, acceptanceDate: new Date() }, latest)
+        this.$store.commit('setCurrentGroupId', this.$route.query.groupId)
         await backend.subscribe(this.$route.query.groupId)
         await this.$store.dispatch('syncContractWithServer', this.$route.query.groupId)
-        let latest = await db.recentHash(this.$route.query.groupId)
-        let acceptance = new Events.AcceptInvitation({ username: this.$store.state.loggedIn, acceptanceDate: new Date() }, latest)
-        this.$store.commit('setCurrentGroupId', this.$route.query.groupId)
         await backend.publishLogEntry(this.$route.query.groupId, acceptance)
-        this.$store.dispatch('deleteMessage', this.$route.query.inviteId)
+        let identityContractId = await namespace.lookup(this.contract.founderUsername)
+        await backend.subscribe(identityContractId)
+        await this.$store.dispatch('syncContractWithServer', identityContractId)
+        this.$store.commit('deleteMessage', this.$route.query.inviteHash)
```

Comment:
Holy crap what is going on in this function?

I saw a really great blog post on good commenting but I can't find it right now, otherwise would link it.

The gist was, add comments like headings to code, so that as you can read them to get a quick overview of what the code is doing, and space them out appropriately. e.g.:

```
// comment about what this is trying to accomplish
code code code;
code code code;
code code code;
code code code;

// comment about what we're going to do next
code code code;
code code code;
code code code;
```

In effect, the comments should serve as both the pseudo code and the heading/commentary about what's going on.

Having one massive chunk of commentless code like this makes it really difficult for newcomers (and reviewers like me) to reason about what is being accomplished in dense chunks like this